src/caching_key_wrapper.rs

```diff
@@ -11,17 +12,19 @@ use crate::key_provider::DataKey;
 use crate::KeyProvider;
 
 #[derive(ZeroizeOnDrop)]
-struct CachedEncryptionEntry {
+struct CachedEncryptionEntry<S: ArrayLength<u8>> {
     #[zeroize(skip)]
     created_at: Instant,
-    key: DataKey,
+    #[zeroize(skip)]
+    key: DataKey<S>,
```

Comment:
For some reason, when I add the generic key size to DataKey<S>, it doesn't want to to work with `ZeroizeOnDrop`